sdk/go/asymmetric_decryopt.go

```diff
@@ -0,0 +1,87 @@
+package ehsm
+
+import (
+	hmac "crypto/hmac"
+	sha256 "crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	ioutil "io/ioutil"
+	"strconv"
+	"time"
+
+	"github.com/iancoleman/orderedmap"
+)
+
+type ADecryptResponse struct {
+	Code    int    `json:"code"`
+	Message string `json:"message"`
+	Result  struct {
+		Plaintext string `json:"plaintext"`
+	} `json:"result"`
+}
+
+func (c *Client) AsymmetricDecrypt(keyid, data, padding_mode string) (string, error) {
+	payload := orderedmap.New()
+	if keyid != "" {
+		payload.Set("keyid", keyid)
+	} else {
+		return "", fmt.Errorf("keyid is empty.")
+	}
+	if padding_mode != "" {
+		payload.Set("padding_mode", padding_mode)
+	} else {
+		return "", fmt.Errorf("padding_mode is empty.")
+	}
+
+	payload.Set("plaintext", data)
+
+	params := orderedmap.New()
+	params.Set("appid", c.appid)
+	params.Set("payload", payload)
+	params.Set("timestamp", strconv.FormatInt(time.Now().UnixNano()/int64(time.Millisecond), 10))
+
+	signString := paramsSortStr(params)
+	hmacSha256 := hmac.New(sha256.New, []byte(c.apikey))
+	hmacSha256.Write([]byte(signString))
+	sign := base64.StdEncoding.EncodeToString(hmacSha256.Sum(nil))
```

Comment:
line39 - line 49 is duplicated for each interfaces, can we split it into a function?

Reply:
ok